crypto/src/ed25519.rs

```diff
@@ -40,12 +46,17 @@ pub struct Ed25519KeyPair {
 
 #[serde_as]
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-pub struct Ed25519Signature(#[serde_as(as = "Ed25519Sig")] pub ed25519_dalek::Signature);
+pub struct Ed25519Signature {
+    #[serde_as(as = "Ed25519Sig")]
+    pub sig: ed25519_consensus::Signature,
+    #[serde(skip)]
+    pub bytes: OnceCell<[u8; ED25519_SIGNATURE_LENGTH]>,
+}
```

Comment:
unfortunately the consensus library does not support as_ref out of the box and had to use this workaround. 

@huitseeker @punwai do you know of any trick to use serder_as to flatten a field to not include the tag? so the serialization would be 
"\"39+HHD6kbDZoFVkDiob1cfHIoq3FE8tMGckMIs58za/lys/omQtvxouGLvJ1SPiLBf7Kge2a4L7uYY1L4iAwCA==\""
instead of
"{\"sig\":\"39+HHD6kbDZoFVkDiob1cfHIoq3FE8tMGckMIs58za/lys/omQtvxouGLvJ1SPiLBf7Kge2a4L7uYY1L4iAwCA==\"}"

otherwise, this would have downstream impact on config files again

Reply:
Yeah, I think you should be using serde_with for this.

Reply:
Not that I've seen - I've just been writing it without serde_as in situations like these.